subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_pn.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/settings/settings.h>
+
+#include "fp_storage_pn.h"
+
+#define SETTINGS_PN_SUBTREE_NAME "fp_pn"
+#define SETTINGS_PN_KEY_NAME "pn"
+#define SETTINGS_NAME_SEPARATOR_STR "/"
+#define SETTINGS_PN_FULL_NAME \
+	(SETTINGS_PN_SUBTREE_NAME SETTINGS_NAME_SEPARATOR_STR SETTINGS_PN_KEY_NAME)
+
+BUILD_ASSERT(SETTINGS_NAME_SEPARATOR == '/');
+
+static char personalized_name[FP_STORAGE_PN_BUF_LEN] = CONFIG_BT_FAST_PAIR_STORAGE_EXT_PN_DEFAULT;
+
+static int settings_set_err;
+static atomic_t settings_loaded = ATOMIC_INIT(false);
+
+static int fp_settings_load_pn(size_t len, settings_read_cb read_cb, void *cb_arg)
+{
+	int rc;
+
+	if (len > sizeof(personalized_name)) {
+		return -EINVAL;
+	}
+
+	rc = read_cb(cb_arg, personalized_name, len);
+	if (rc < 0) {
+		return rc;
+	}
+
+	if (rc != len) {
+		return -EINVAL;
+	}
+
+	if (strnlen(personalized_name, len) != (len - 1)) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int fp_settings_set(const char *name, size_t len, settings_read_cb read_cb, void *cb_arg)
+{
+	int err = 0;
+	const char *key = SETTINGS_PN_KEY_NAME;
+	static bool name_loaded;
```

Comment:
Do we need to check if PN was loaded? Settings should provide only the last stored value. See API doc:https://docs.zephyrproject.org/latest/services/settings/index.html#c.settings_store_itf.csi_load